icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java

```diff
@@ -4436,6 +4448,26 @@ public void TestAccountingCurrency() {
         }
     }
 
+    /**
+     * en_ID/sh_ME uses language only locales en/sh which requires NumberFormatServiceShim to fill in the currency, but
+     * prior to ICU-20631, currency was not filled in for accounting, cash and standard, so currency placeholder was
+     * used instead of the desired locale's currency.
+     */
+    @Test
+    public void TestCurrencyFormatForMissingLocale() {
+        ULocale loc = new ULocale("sh_ME");
+        Currency cur = Currency.getInstance(loc);
+        NumberFormat curFmt = NumberFormat.getInstance(loc, NumberFormat.CURRENCYSTYLE);
+        assertNotNull("NumberFormat is missing currency instance for CURRENCYSTYLE", curFmt.getCurrency());
+        assertEquals("Currency instance is not for the desired locale for CURRENCYSTYLE", cur, curFmt.getCurrency());
```

Comment:
For each case, can you also add one more assertEquals please,

```java
assertEquals("NumberFormat produced output with currency symbol", "100.00 €", cufFmt.format(100));
```

for whatever the output is.  Just to be sure.

Reply:
Added suggestion.